DNN Platform/Library/Obsolete/HostController.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.Entities.Controllers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using DotNetNuke.Common;
+    using DotNetNuke.ComponentModel;
+
+    using Microsoft.Extensions.DependencyInjection;
+
+    using INewHostController = DotNetNuke.Abstractions.Entities.Controllers.IHostController;
+
+    // None of the APIs are deprecated, but the IHostController
+    // is deprecated and moved to the abstractions project. When
+    // it is time to remove APIs we should remove the parent
+    // classes listed here
+
+    /// <inheritdoc />
+    public partial class HostController : ComponentBase<IHostController, HostController>, IHostController
+    {
+        [Obsolete("Deprecated in 9.7.1. Scheduled for removal in v11.0.0, use DotNetNuke.Abstractions.Controllers.IHostController instead.")]
+        public static new IHostController Instance
+        {
+            get
+            {
+                var newHostController = Globals.DependencyProvider.GetRequiredService<INewHostController>();
```

Comment:
I updated this from the initial PR submission to remove the static instance. This created a poorly implemented singleton that prevented the Dependency Injection container from operating correctly. We should offload calls like this to use the container as it will manage instance state for us automatically. If we want the `IHostController` to be a singleton for example, let's leave that responsibility to the container.